js/models/conversations.js

```diff
@@ -1475,30 +1487,77 @@
       return !this.get('left');
     },
 
-    async endSession() {
-      if (this.isPrivate()) {
-        const now = Date.now();
-        const message = this.messageCollection.add({
-          conversationId: this.id,
-          type: 'outgoing',
-          sent_at: now,
-          received_at: now,
-          destination: this.id,
-          recipients: this.getRecipients(),
-          flags: textsecure.protobuf.DataMessage.Flags.END_SESSION,
+    async onSessionResetInitiated() {
+      if (this.get('sessionResetStatus') === SessionResetEnum.none) {
+        this.set({ sessionResetStatus : SessionResetEnum.initiated });
+        await window.Signal.Data.updateConversation(this.id, this.attributes, {
+          Conversation: Whisper.Conversation,
         });
```

Comment:
Maybe make a function similar to the one I did for the friend request status?
```
async setSessionResetStatus(newStatus) {
  // Ensure that the new status is a valid SessionResetEnum value
  if (!Object.values(SessionResetEnum).some(v => v === newStatus))
    return;
  if (this.get('sessionResetStatus') !== newStatus) {
    this.set({ sessionResetStatus: newStatus });
    await window.Signal.Data.updateConversation(this.id, this.attributes, {
      Conversation: Whisper.Conversation,
    });
  }
}
```

Reply:
Good idea.
I've just replaced the `array.some` with `... in ...` for readability :)